db/src/renumber.rs

```diff
@@ -300,10 +300,16 @@ mod tests {
                             [268435458 :db/txInstant ?ms 268435458 true]]"#);
 
         // N.b., after renumbering, the on-disk data is not reflected in the `TestConn`!
-        let mut db = renumber(&local.sqlite,
+        let mut db;
```

Comment:
I think something has gotten confused.  Everything here is going to take place inside an encapsulating SQL transaction; we shouldn't be managing transactions in this helper function at all.  It's up to consumers to manage isolation, etc; and `sync` (like `transact`) will do so.

Reason to yourself: suppose renumbering succeeds but something later fails -- how will I roll back renumbering?  If you commit, you can't.  (Without using nested transactions or checkpoints or other complicated things, which I really don't think we need.)